spec/connection_spec.rb

```diff
@@ -50,13 +52,18 @@
       settings = SETTINGS.dup
       settings[:payload] = [[:settings_header_table_size, 256]]
 
-      expect(@conn).to receive(:send) do |frame|
-        expect(frame[:type]).to eq :settings
-        expect(frame[:flags]).to eq [:ack]
-        expect(frame[:payload]).to eq []
+      # We expect two frames here - one for the connection setup, and one for the settings ack.
+      frames = []
+      expect(@conn).to receive(:send).twice do |frame|
+        frames << frame
       end
 
       @conn << f.generate(settings)
+
+      frame = frames.last
```

Comment:
We should also check that first frame is the preface?

Reply:
Yeah, why not? It's good.